README.md

```diff
@@ -2,16 +2,171 @@
+
+## What is a product runtime?
+
+A product runtime is a logical grouping of a set of features/components that creates a virtual boundary for execution. Every Carbon product has a default runtime shipped with it. With the Carbon feature plugin, you can create runtimes and install Carbon features to that runtime.
+
+A Carbon product (WSO2 ESB, WSO2 AS etc.) is made by installing the relevant features on top of the Carbon kernel using the following steps:
+
+1. Create a runtime on top of the Carbon kernel using the [generate-runtime Maven goal](#configuring-the-generate-runtime-maven-goal). 
+2. Install the relevant features using the [`install` Maven goal](#configuring-the-install-maven-goal).
+
+## Carbon feature plugin configurations
+
+The following sections describe the configurations of the Maven goals that are implemented in the Carbon feature plugin. 
+
+### Configuring the generate Maven goal
+
+A sample pom.xml file configuration of the `generate` Maven goal is shown below.
+
+            <feature>
+               <id>org.wso2.ciphertool.feature.group</id>
+               <version>4.4.0</version>
+            </feature>
+        </features>
+ 
+ > Add the ID of the feature being installed as the value of the `<ID>` property, and add the version of the feature being installed as the value of the `<version>` property.
+ 
+* `deleteOldRuntimeFiles`: Whether to delete old `*.profile` folders located in the `<CARBON_HOME>/repository/components/p2/org.eclipse.equinox.p2.engine/profileRegistry/` directory. The default value is set to true.
+
+ NOT MANDATORY.
+ Example: `<deleteOldRuntimeFiles>false</deleteOldRuntimeFiles>`.
+
+### Configuring the uninstall Maven goal
+
+A sample pom.xml file configuration of the uninstall Maven goal is shown below.
+
+      <build>
+    	<plugins>
+        	<plugin>
+            	<groupId>org.wso2.carbon.maven</groupId>
+            	<artifactId>carbon-feature-plugin</artifactId>
+            	<version>${carbon.feature.plugin.version}</version>
+            	<executions>
+                	<execution>
+                    	<id>p2-feature-generation</id>
+                    	<phase>package</phase>
+                    	<goals>
+                        	<goal>uninstall</goal>
+                    	</goals>
+                    	<configuration>
+                        	//plugin configuration goes here.
+                    	</configuration>
+                	</execution>
+            	</executions>
+        	</plugin>
+            </plugins>
+          </build>
+	  
+You can modify the above file to add the configurations of the plugin by adding the following parameters within the `<configuration>` element of it. 
+
+* `destination`: Points to the `<CARBON_HOME>/repository/components/` directory.
+
+ MANDATORY property.
+ Example: `<destination>/home/Carbon/wso2carbon-4.4.0/repository/components</destination>`
+ 
+* `runtime`: Which runtime in the destination from where you need to uninstall the features. 
```

Comment:
*The runtime required (in the destination from where you need to uninstall the features).